lib/accumulate_distribute/events/orders_order_fill.js

```diff
@@ -23,14 +24,14 @@ const scheduleTick = require('../util/schedule_tick')
  */
 const onOrdersOrderFill = async (instance = {}, order) => {
   const { state = {}, h = {} } = instance
-  const { args = {}, ordersBehind, timeout, currentOrder } = state
+  const { args = {}, ordersBehind, timeout, currentOrder, minDistortedAmount } = state
   const { emit, updateState, debug } = h
   const { catchUp } = args
 
   const newOrdersBehind = Math.max(0, ordersBehind - 1)
   const fillAmount = order.getLastFillAmount()
-  const remainingAmount = state.remainingAmount - fillAmount
-  const absRem = Math.abs(remainingAmount)
+  const remainingAmount = nBN(state.remainingAmount).minus(fillAmount).toNumber()
+  const absRem = nBN(state.remainingAmount).abs().minus(Math.abs(fillAmount)).toNumber()
```

Comment:
I would use `nBN(fillAmount).abs()` instead of `Math.abs(fillAmount)` to be sure that it works even if on future we receive this amount as str by mistake

Reply:
good catch .. done :)